tests/integration/class-s3-media-sync-test.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ * Integration tests for S3_Media_Sync
+ *
+ * @package S3_Media_Sync
+ */
+
+namespace S3_Media_Sync\Tests\Integration;
+
+use S3_Media_Sync;
+
+/**
+ * S3_Media_Sync test case.
+ *
+ * @covers S3_Media_Sync
+ */
+class S3_Media_Sync_Test extends WP_Integration_Test_Case {
+
+	private array $complete_settings
+		= [
+			'bucket'     => 'test-bucket',
+			'key'        => 'test-key',
+			'secret'     => 'test-secret',
+			'region'     => 'test-region',
+			'object_acl' => 'public-read',
+		];
+
+	private array $incomplete_settings
+		= [
+			'bucket'     => '',
+			'key'        => '',
+			'secret'     => '',
+			'region'     => '',
+			'object_acl' => 'public-read',
+		];
+
+	/**
+	 * Helper method for that checking media sync hooks are registered.
+	 *
+	 * @param string $assertion The assertion to use.
+	 */
+	private function assert_media_syncs_hooks_registered( string $assertion ): void {
+		self::$assertion( has_filter( 'wp_handle_upload', [ $this->s3_media_sync, 'add_attachment_to_s3' ] ) );
+		self::$assertion( has_action( 'delete_attachment', [ $this->s3_media_sync, 'delete_attachment_from_s3' ] ) );
+		self::$assertion( has_filter( 'wp_save_image_editor_file', [ $this->s3_media_sync, 'add_updated_attachment_to_s3' ] ) );
+	}
+
+	/**
+	 * @testdox Allow media syncs when required settings set
```

Comment:
I think if you call phpunit with `--testdox`, then this line which duplicates the method name isn't needed. The `--testdox` flag will take the method name, strip a `test_` prefix, and change underscores to spaces anyway.

Reply:
If a test class method can't be made into a human-readable sentence, such that the `@testdox` would be needed, then sure, use it, but I'd question why the test class method couldn't be rewritten to make it a human-readable sentence in the first place.